### DISCUSSION TITLE
Support for the (AP) version of these modules

### DISCUSSION THREAD

**Discussion by brcisna** [iamromulan/cellular-modem-wiki#80] [Ideas/Requests - Quectel]
Is there any chance of integrating support for the (AP) version of the RM520N-GL modules.
This is the PCIe only iteration of these modules.
We never have determined conclusively if the AA firmware will work on these(AP)  modules.

TIA

**Comment by iamromulan** [comment]
I started to look at this but never revisited as I could never get a straight answer from Quectel about this. I think certain AP units are able to be modified to become AA. Great example: https://www.rework.network/collections/lte-home-gateway/products/5g2phy-rm520

I will look at adding the PCIe drivers to QuecDeploy. I have used this after installing the drivers to utilize it over PCIe with my windows desktop.
https://a.aliexpress.com/_mqoeBGr

**Comment by brcisna** [comment]
Edit: Failed to mention. I was trying this on Debian Trixie 13, kernel 6.9,,, a couple months ago, 
Also worth mentioning this card would never show in its own IOMMU group so i could not pass it through to my Windows 10 VM to try it on Windows. Even with the scratch built Quextel mhi_ driver made,, into Linux.  I think that is part of the problem these are so sparadic.  It combined with say the next pcie device next to it,in the adjoining pcie slot of the IOMMU group. 

Thank You for the info Cameron, If anyone can figure it out you can!
The AP module i got from AliX,,,all i could ever see was one interface
/dev/hmi_??. Hate to say cant even remember what was after the hmi_,,,was
three or four uppercase letters. That is the only device that appeared
I have the exact same adapter you linked. I built the actual mhi driver
from Quectel, and seems like the device name changed slightly in lspci. I
tried everything i knew of,, and just couldnt get any more devices to
show,,, no /dev/usbTTY, which i dont think this card would generate anyway?
,,,or serial stuff.
I finally even hooked up the usb cable to my card to see if any usb
showed,,and no usb info appeared for the  Ap module,,,, just to make
sure,,i wasnt missing something. I thought possibly the usb power had to be
supplied to get the card to work for whatever reason.
Really wanted to try and get that one to work,,but finally gave up and sent
it back.

Barry

**Comment by M0rvius** [comment]
> QuecDeploy

hello, thank you for the work done, please let me know if you can run RM520N-GLAP in PCIe EP mode for use with the Ethernet chipset "eth_driver", that would be great news

**Comment by brcisna** [comment]
Hi Cameron,

Hope things are going well.

Just curious if you ever delved into the Quectel RM520N-GLAP anymore.....

I did purchase one of these several months ago,,,and messed with it
some,,,I never could get it to responde to most AT commands.... I have the
M.2 B-Key to PCIe x1 adapter,,,that does both PCIe and USB 3.0.
I finally got discouraged and end up sending back to Ali... I would really
like to make this work in PCIe mode

By the way,,,I did get the Fibocom,, FM350-GL....to work in PCIe mode
fine,,,,It is a  great modem,,,bang for the buck,,, It worked out of the
box,,,under Debian Trixie  with latest modem manager and I install the
ModemManager-GUI... All i had to do is create a  connection,,,,in
nm-connection editor gui for the T-MO APN,,,and it works
great,,,,,couldnt believe it..
It is very comparable to the RM520n-GL if you bevcer looked in ti ti,,it
also has built in esim,,,,which is nice...

Thanks,
Barry
